modules/tensor_mechanics/include/actions/TensorMechanicsAction.h

```diff
@@ -23,17 +23,32 @@ class TensorMechanicsAction : public Action
 
 protected:
   virtual std::string getKernelType();
-  virtual InputParameters getParameters(std::string type);
+  virtual InputParameters getKernelParameters(std::string type);
 
+  ///@{ displacement variables
   std::vector<NonlinearVariableName> _displacements;
   unsigned int _ndisp;
-
   std::vector<VariableName> _coupled_displacements;
+  ///@}
 
+  ///@{ residual debugging
   std::vector<AuxVariableName> _save_in;
   std::vector<AuxVariableName> _diag_save_in;
+  ///@}
 
   Moose::CoordinateSystemType _coord_system;
+
+  /// if this vector is not empty the variables, kernels and materials are restricted to these subdomains
+  std::vector<SubdomainName> _subdomain_names;
+
+  /// strain formulation
+  enum class Strain { Small, Incremental, Finite } _strain;
```

Comment:
Per our discussion, because small/finite and incremental/total are orthogonal, I think we should have two separate enums or booleans here: one to determine whether it's small or finite strain, and one to determine whether we compute a total or incremental strain.